src/groups.py

```diff
@@ -20,9 +19,11 @@ def draw(self, target_pos: Coordinate):
 
         for layer in LAYERS.values():
 
-            for sprite in sorted(
-                    self.sprites(),
-                    key=lambda spr: spr.rect.centery):
+            if layer == LAYERS["main"]:
+                sorted_sprites = sorted(self.sprites(), key=lambda sprite: sprite.hitbox_rect.centery)
+            else:
+                sorted_sprites = sorted(self.sprites(), key=lambda sprite: sprite.rect.centery)
+
+            for sprite in sorted_sprites:
                 if sprite.z == layer:
-                    self.display_surface.blit(
-                        sprite.image, sprite.rect.topleft + (self.offset if not isinstance(sprite, TextBox) else (0, 0)))
+                    sprite.draw(self.display_surface, self.offset)
```

Comment:
Removal of the offset-independent drawing for textboxes creates a bug.

Reply:
Alright, will add it back 😄

Reply:
I think it will be clearer to add a draw method in TextBox class and draw it without an offset. I added a draw method to draw apples and the associated tree at the same time for instance (to resolve layer conflicts)

Reply:
That makes only three instances of that behaviour in the entire code. Everything else has standardised drawing code. For now, nothing justifies having to split that apart. Also, as for layer conflicts, nothing actually prevents us from using `pygame.sprite.LayeredUpdates` as I suggested a while ago

Reply:
In fact, the apples should not work with a layer. I don’t want to display the trees and then the apples; otherwise, we’ll have a problem where the player will either be below or above them. We need to blit the apples associated with the tree when we blit a tree. And there’s no need to add a draw method for each class; it’s enough to put one in the Sprite class.

Reply:
> otherwise, we’ll have a problem where the player will either be below or above them.

Would you mind explaining why? To me this is not a problem, as if the player is actually above the tree they will be displayed behind, and they can't get close enough to the apples from the bottom anyway